execute/row_fn.go

```diff
@@ -13,52 +12,85 @@ import (
 )
 
 type dynamicFn struct {
-	compilationCache *compiler.CompilationCache
-	inRecord         values.Object
-
-	preparedFn compiler.Func
-
+	// Configuration attributes. These are initialized once
+	// on creation and used for each new compilation.
+	scope      compiler.Scope
+	fn         *semantic.FunctionExpression
 	recordName string
-	record     *Record
 
-	recordCols map[string]int
-	references []string
+	// These are initialized after the function is prepared.
+	preparedFn compiler.Func
+	arg0       values.Object
+	args       values.Object
```

Comment:
What are `args` and `args0`? Why do we need both?

Reply:
`args` is the object for calling the function and `arg0` is the first argument specifically in the constructed object.

Arguments are sent to functions by using an object expression so `foo({a: 1, b: 2})` is two arguments, but the `Eval` function only takes the argument for the first one. So `arg0` is the object for `r` itself while `args` is for the all parameters.